Update multiple transports information (#73)
smartdevicelink/sdl_core_guides: docs/Feature Documentation/Multiple Transports/index.md

# General Description

> The Multiple Transports feature allows apps connected to Core to start another connection over a different transport for certain services (for example, an app connected over bluetooth using WiFi as a secondary transport for video streaming). This guide will give an overview of the process which is used to establish a secondary transport connection. See [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md) for more details on the original feature proposal.

Reviewer: ```suggestion
The Multiple Transports feature allows apps connected to SDL Core to start another connection over a different transport for certain services. For example, an app connected over bluetooth could use WiFi as a secondary transport for video streaming. This guide will give an overview of the configuration settings for establishing a secondary transport connection. See [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md) for more details on the original feature proposal.
```

> The Multiple Transports feature allows apps connected to Core to start another connection over a different transport for certain services (for example, an app connected over bluetooth using WiFi as a secondary transport for video streaming). This guide will give an overview of the process which is used to establish a secondary transport connection. See [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md) for more details on the original feature proposal.

Reviewer: ```suggestion
The Multiple Transports feature allows apps connected to SDL Core to start another connection over a different transport for certain services. For example, an app connected over bluetooth may use WiFi as a secondary transport for video streaming. This guide will give an overview of the process which is used to establish a secondary transport connection. See [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md) for more details on the original feature proposal.
```

Reviewer 2: I think "Bluetooth" and "Secondary Transport" should be capitalized.  

I'd also change the proposal reference to "See [SDL 0141 - Supporting Simultaneous Multiple Transports](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md) for more..."

## Implementation
- After the proxy is connected to Core, it initiates another connection over a different transport.
- Core tells the proxy which transport can be used as Secondary Transport.
- The services that are allowed on the Secondary Transport are specified by Core.

!!! Note

> RPC and Hybrid services only run on the Primary transport

Reviewer 2: ```suggestion
RPC and Hybrid services only run on the Primary Transport
```

!!!

> There are three protocol control frames which are used in the implementation of multiple transports

Reviewer 2: ```suggestion
There are three protocol control frames which are used in the implementation of multiple transports.
```

Reviewer 2: @jacobkeeler it looks like adding the period here was missed in your last commit.

### StartService ACK
```

> {

Reviewer: Can you add a code comment that this is the payload?

>     "audioServiceTransports" : [1, 2],

Reviewer: Can you provide some context what these parameters are for? StartService Ack payload?

    "videoServiceTransports" : [1, 2],
    "secondaryTransports" : ["TCP_WIFI"]
}
```

> Core responds to the proxy's `StartService` request with additional parameters audioServiceTransports, videoServiceTransports and secondaryTransports.   

Reviewer 2: ```suggestion
Core responds to the proxy's `StartService` request with additional parameters `audioServiceTransports`, `videoServiceTransports`, and `secondaryTransports`.   
```

> - The `secondaryTransports` contains an array of the allowed secondary transports for the current Primary transport. 

Reviewer 2: ```suggestion
- The `secondaryTransports` parameter contains an array of the allowed Secondary Transports for the current Primary Transport. 
```

- `audioServiceTransports` and `videoServiceTransports` describe which services are allowed to run on which transports (Primary=1, Secondary=2, or both). The proxy uses this information and starts services only on allowed transports.
- This response is constructed by Core using the configurations defined in the SDL INI file, described in [this guide](../../getting-started/multiple-transports-configuration).

> - Since RPC and Hybrid services always run on Primary transport, only Video and Audio services are configurable.

Reviewer 2: ```suggestion
- Since RPC and Hybrid services always run on Primary Transport, only Video and Audio services are configurable.
```

### TransportEventUpdate
```
{
    "tcpIpAddress" : "192.168.1.1",
    "tcpPort" : 12345
}
```

> Core sends a TransportEventUpdate to the proxy to provide additional information required to connect over the TCP transport when it is available.

Reviewer 2: ```suggestion
Core sends a `TransportEventUpdate` to the proxy to provide additional information required to connect over the TCP transport when it is available.
```
Not sure if backticks are necessary here or not; I'll leave that up to the your discretion!

> - If the `tcpIpAddress` field is empty, the secondary transport is unavailable and the proxy will not send a RegisterSecondaryTransport request 

Reviewer 2: ```suggestion
- If the `tcpIpAddress` field is empty, the Secondary Transport is unavailable and the proxy will not send a RegisterSecondaryTransport request. 
```

### RegisterSecondaryTransport

> Using the information in the StartService ACK and TransportEventUpdate frames, the proxy sends a RegisterSecondaryTransport request over the secondary transport with the same session ID as the Primary transport.  

Reviewer 2: ```suggestion
Using the information in the StartService ACK and TransportEventUpdate frames, the proxy sends a `RegisterSecondaryTransport` request over the Secondary Transport with the same session ID as the Primary Transport.  
```
"Secondary Transport" and "Primary Transport" are capitalized throughout the accepted Evolution Proposal, so I think we want to keep that formatting here.

> - If Core sends back a RegisterSecondaryTransport ACK, the proxy can start services over the secondary transport

Reviewer 2: ```suggestion
- If Core sends back a RegisterSecondaryTransport ACK, the proxy can start services over the Secondary Transport.
```

## Operation Examples

|||

> Start Service (WiFi as secondary transport)

Reviewer 2: ```suggestion
Start Service (WiFi as Secondary Transport)
```

![StartService](./assets/StartService.png)
|||

|||

> Start Video/Audio service (Over secondary transport)

Reviewer 2: ```suggestion
Start Video/Audio service (Over Secondary Transport)
```
